### PR TITLE
fix: use fs.stat in favor of fs.access

### PR DESCRIPTION
**Pull request by P0lip**
Fixes https://github.com/stoplightio/platform-internal/issues/1509

**Checklist**

- [x] Tests added / updated
- [x] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/rulesets/finder.ts [hunk 1]
```diff
@@ -75,7 +76,7 @@ export async function findFile(from: string, to: string) {
 
 function exists(uri: string): Promise<boolean> {
   return new Promise<boolean>(resolve => {
-    fs.access(uri, fs.constants.F_OK, err => {
+    fs.stat(uri, err => {
```
Comment (nulltoken) [inline src/rulesets/finder.ts:79]: @P0lip Why this change? The doc seems to state _"To check if a file exists without manipulating it afterwards, `fs.access()` is recommended."_

Another approach would be to just use `existsSync`.

Thoughts?

File: src/rulesets/finder.ts [hunk 2]
```diff
@@ -2,7 +2,8 @@ import * as path from '@stoplight/path';
 import * as fs from 'fs';
 import { RESOLVE_ALIASES, STATIC_ASSETS } from '../assets';
 
-const SPECTRAL_SRC_ROOT = path.join(__dirname, '..');
+// let's point at dist directory that has all relevant files (including custom functions) transpiled
+const SPECTRAL_SRC_ROOT = path.join(__dirname, '../../dist');
```
Comment (nulltoken) [inline src/rulesets/finder.ts:6]: @P0lip What led you to switch over to `dist`?

File: src/rulesets/finder.ts (hunk 1 above)
Comment (P0lip) [reply src/rulesets/finder.ts:79]: It's because of studio Web that does not support fs.access.

File: src/rulesets/finder.ts (hunk 2 above)
Comment (P0lip) [reply src/rulesets/finder.ts:6]: `src` isn't the best content root, since it does not have all files transpiled, i.e. custom functions are left as typescript files.
At the time I wrote the code, we didn't have the concept of custom functions, so it didn't really matter since we stored JSON files in both `src` and `dist` directories.
Does it make sense?

Comment (nulltoken) [reply src/rulesets/finder.ts:6]: Perfect sense!

File: src/rulesets/finder.ts (hunk 1 above)
Comment (nulltoken) [reply src/rulesets/finder.ts:79]: Ok.... Although that's out of the scope of that's PR, maybe should we implement some kind of a jest global `fs` mock that fails when the core leverages functions that aren't supported by studio Web?
